### DISCUSSION TITLE
How do we add a certificate to truststore in docker setup?

### DISCUSSION THREAD

**Discussion by cemremengu** [keycloak/keycloak#10506] [Q&A] [answered]
I have searched a lot of places but so far no luck. I am trying to make an LDAPS connection with the **docker** setup of Keycloak but I have no idea where to add the certificate. The installation has no trust store (or at least I couldn't find) and java binaries included in the container are confusing (there are multiple paths like `java-11-openjdk-11.0.14.0.9-2.el8_5.x86_64`,  `jre`,  `jre-11`,  `jre-11-openjdk`,  `jre-11-openjdk-11.0.14.0.9-2.el8_5.x86_64  jre-openjdk`).

What should I do to make Keycloak trust my certificate?

**Comment by nathcoad** [comment, marked as answer]
I also wondered about this. Here's what I ended up doing.

Reference https://www.keycloak.org/server/keycloak-truststore we see that we can pass a java keystore to the docker entry point using the command line arguments `--spi-truststore-file-file=myTrustStore.jks --spi-truststore-file-password=password --spi-truststore-file-hostname-verification-policy=ANY`

So now we just ned to create a java keystore file, which you can find examples of in many places such as https://stackoverflow.com/questions/11952274/how-can-i-create-keystore-from-an-existing-certificate-abc-crt-and-abc-key-fil

So I created my keystore, named it as myTrustStore.jks and put it in a volume that I mapped to the docker container. Then I modified the command line in my docker-compose.yml file to something like this:

`command: start --auto-build --spi-truststore-file-file=/etc/x509/myTrustStore.jks --spi-truststore-file-password=change_me --spi-truststore-file-hostname-verification-policy=ANY`
